### PR TITLE
[Conformance][TorchFX] GPU quantization support

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

* --validate-in-backend CLI option is added
* CUDA_FX_TORCH backend is added to conformance test
* FXSQMultiply is updated to work on both CPU and GPU

### Tests

Local run:
CLI: `python -m pytest test_quantize_conformance.py -k CUDA_FX --data path/to/imagenet`

| Model                             | Backend       | Metric name | Metric value | Metric diff | Num FQ | Num int4 | Num int8 | Compr. time | Total time | RAM MiB | Status | Build url |
|-----------------------------------|---------------|-------------|--------------|-------------|--------|----------|----------|-------------|------------|---------|--------|-----------|
| torchvision/resnet18              | CUDA_FX_TORCH | Acc@1       | 0.6942       | -0.0036     | 30     | 0        | 21       | 0:00:02     | 0:04:14    | 1560    |        |           |
| torchvision/swin_v2_s             | CUDA_FX_TORCH | Acc@1       | 0.83572      | -0.0014     | 149    | 0        | 101      | 0:00:55     | 0:17:24    | 3161    |        |           |
| torchvision/vit_b_16              | CUDA_FX_TORCH | Acc@1       | 0.80962      | -0.00108    | 62     | 0        | 50       | 0:00:19     | 0:13:39    | 2876    |        |           |
| torchvision/mobilenet_v3_small_BC | CUDA_FX_TORCH | Acc@1       | 0.66642      | -0.01018    | 61     | 0        | 36       | 0:00:05     | 0:04:09    | 1653    |        |           |

### CODE REVIEW

File: nncf/quantization/algorithms/smooth_quant/torch_fx_backend.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
 class FXSQMultiply(torch.nn.Module):
     def __init__(self, scale: torch.Tensor):
         super().__init__()
-        self._scale_value = scale
+        self._scale_value = torch.nn.Parameter(scale)
```
Comment (alexsu52) [inline nncf/quantization/algorithms/smooth_quant/torch_fx_backend.py:45]: I would suggest to Buffer instead of Parameter.

Comment (daniil-lyakhov) [reply nncf/quantization/algorithms/smooth_quant/torch_fx_backend.py]: Done

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

File: nncf/experimental/torch/fx/constant_folding.py [hunk 2]
```diff
@@ -235,6 +235,20 @@ def _is_impure(node: torch.fx.Node) -> bool:
     return node.op in {"placeholder", "output"}
 
 
+def get_model_device(model: torch.fx.GraphModule) -> torch.device:
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/constant_folding.py]: Done

File: tests/post_training/conftest.py [hunk 3]
```diff
@@ -19,6 +19,11 @@ def pytest_addoption(parser):
     parser.addoption("--fp32", action="store_true", help="Test original model")
     parser.addoption("--cuda", action="store_true", help="Enable CUDA_TORCH backend")
     parser.addoption("--benchmark", action="store_true", help="Run benchmark_app")
+    parser.addoption(
+        "--validate-in-backend",
+        action="store_true",
+        help="Validate quantized model in native backend, not in openvino.",
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/post_training/conftest.py]: Fixed, please check

File: tests/torch/fx/test_model_transformer.py [hunk 4]
```diff
@@ -547,6 +548,21 @@ def test_constant_folding():
     check_graph(nncf_graph, "folded_model.dot", TRANSFORMED_GRAPH_DIR_NAME, extended=True)
 
 
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="Cuda is not available")
+def test_constant_folding_scalar_clone():
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/fx/test_model_transformer.py]: Done

File: nncf/experimental/torch/fx/constant_folding.py [hunk 5]
```diff
@@ -251,9 +265,11 @@ def constant_fold(
             cf = ConstantFolder(gm)
             cf.run()
 
+            device = get_model_device(gm)
             for node, constant in cf.node_replacements.items():
                 if constraint_fn is not None and not constraint_fn(node):
                     continue
+                constant = constant.to(device)
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/constant_folding.py]: Unfortunately, for some nodes (like cloned scalar from this model), the device in the meta differs from the actual model device https://github.com/openvinotoolkit/nncf/pull/3010/files#diff-8e3f87fb950bb27e4bbc8dae723516a9178525675fed9f9d5f8ef2f304c0b27fR639

**Review by AlexanderDokuchaev — APPROVED** [review]
